### PR TITLE
Move shareable callbacks for fetching search query summary to Convers…

### PR DESCRIPTION
**Pull request by yrliou**
…ationDriver

Resolves https://github.com/brave/brave-browser/issues/40741

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
should be covered by auto tests

### CODE REVIEW

File: components/ai_chat/core/browser/conversation_driver.cc [hunk 1]
```diff
@@ -345,43 +370,97 @@ bool ConversationDriver::ShouldFetchSearchQuerySummary() {
          should_send_page_contents_;
 }
 
-void ConversationDriver::MaybeFetchOrClearSearchQuerySummary() {
+void ConversationDriver::MaybeFetchOrClearSearchQuerySummary(
+    FetchSearchQuerySummaryCallback callback) {
```
[the remark this answers is not in the document]
Comment (petemill) [reply components/ai_chat/core/browser/conversation_driver.cc:374]: Actually I don't see any calls where this param is provided, apart from the test. Is it needed?

**Review by petemill — COMMENTED** [review]
Everything looks good, but just a small question

File: components/ai_chat/core/browser/conversation_driver.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (yrliou) [reply components/ai_chat/core/browser/conversation_driver.cc:374]: It's needed unfortunately for test, in the browser tests, it won't wait for the operation on renderer process unless we use a callback to explicitly quit a run loop on the main thread.

[the remark this answers is not in the document]
Comment (yrliou) [reply components/ai_chat/core/browser/conversation_driver.cc:374]: Regarding the default value, I would argue a default value in declaration (.h) is clear tho.

[the remark this answers is not in the document]
Comment (petemill) [reply components/ai_chat/core/browser/conversation_driver.cc:374]: I've been asked to change this multiple times in reviews for my PRs, so now I'm used to doing it. We can consult the style guide but if it doesn't mention it then I'll let you choose what you think is best.

**Review by petemill — APPROVED** [review]

File: components/ai_chat/core/browser/conversation_driver.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (yrliou) [reply components/ai_chat/core/browser/conversation_driver.cc:374]: Not seeing it's mentioned specifically in the style guide, I'm merging as is, but can always add a comment later if needed, thanks.

[the remark this answers is not in the document]
Comment (petemill) [reply components/ai_chat/core/browser/conversation_driver.cc:374]: I agree, I can't see anything either and there're plenty examples of not doing that in the codebase, so it's all ok

**Pull request merged**